src/main/java/com/jpr/maintenance/database/repository/UserRepository.java

```diff
@@ -0,0 +1,22 @@
+package com.jpr.maintenance.database.repository;
+
+import com.jpr.maintenance.database.model.UserEntity;
+import org.springframework.data.r2dbc.repository.Modifying;
+import org.springframework.data.r2dbc.repository.Query;
+import org.springframework.data.repository.reactive.ReactiveCrudRepository;
+import org.springframework.stereotype.Repository;
+import reactor.core.publisher.Mono;
+
+@Repository
+public interface UserRepository extends ReactiveCrudRepository<UserEntity, Long> {
+    @Query("SELECT * FROM \"user\" WHERE username = :username AND \"password\" = :password")
+    Mono<UserEntity> findByUserNameAndPassword(String username, String password);
+
+    @Modifying
+    @Query("INSERT INTO \"user\" (username, \"password\") VALUES (&username, &password)")
+    Mono<Integer> saveUser(String username, String password);
```

Comment:
true, its because i had to use the @query and @modifying 
from the docs i got:
> The result of a modifying query can be:
> Void (or Kotlin Unit) to discard update count and await completion.
> Integer or another numeric type emitting the affected rows count.
> Boolean to emit whether at least one row was updated.

Could of course combine this with a select query in the 'UserService' to still return the entity, what do you think?

Reply:
I propose that we make it our convention after inserting a value that we return the primary key. We could do that using the "RETURNING" keyword in [Postgres](https://www.postgresql.org/docs/14/dml-returning.html). 

From that page, here is an example of the syntax:
`
CREATE TABLE users (firstname text, lastname text, id serial primary key);

INSERT INTO users (firstname, lastname) VALUES ('Joe', 'Cool') RETURNING id;
`